starter-code/pizza.js

```diff
@@ -17,69 +17,187 @@ var state = {
   greenPeppers: true,
   whiteSauce: false,
   glutenFreeCrust: false
-}
+};
 
 // This function takes care of rendering the pizza based on the state
 // This function is triggered once at the begining and everytime the state is changed
 function renderEverything() {
-  renderPepperonni()
-  renderMushrooms()
-  renderGreenPeppers()
-  renderWhiteSauce()
-  renderGlutenFreeCrust()
-
-  renderButtons()
-  renderPrice()
+  renderPepperonni();
+  renderMushrooms();
+  renderGreenPeppers();
+  renderWhiteSauce();
+  renderGlutenFreeCrust();
+
+  renderButtons();
+  renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll(".pep").forEach(function($pep) {
     if (state.pepperonni) {
       $pep.style.visibility = "visible";
-    }
-    else {
+    } else {
       $pep.style.visibility = "hidden";
     }
-  })
+  });
 }
 
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll(".mushroom").forEach(function($mushroom) {
+    if (state.mushrooms) {
+      $mushroom.style.visibility = "visible";
+    } else {
+      $mushroom.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderGreenPeppers() {
-  // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll(".green-pepper").forEach(function($greenPeppers) {
+    if (state.greenPeppers) {
+      $greenPeppers.style.visibility = "visible";
+    } else {
+      $greenPeppers.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderWhiteSauce() {
-  // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll(".sauce").forEach(function($sauce) {
+    if (state.whiteSauce) {
+      $sauce.classList.add("sauce-white");
+    } else {
+      $sauce.classList.remove("sauce-white");
+    }
+  });
 }
 
 function renderGlutenFreeCrust() {
-  // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  document.querySelectorAll(".crust").forEach(function($gfc) {
+    if (state.glutenFreeCrust) {
+      $gfc.classList.add("crust-gluten-free");
+      // $gfc.style.visibility = "visible";
+    } else {
+      $gfc.classList.remove("crust-gluten-free");
+      // $gfc.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderButtons() {
-  // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  document.querySelectorAll(".btn-pepperonni").forEach(function($btn) {
```

Comment:
instead of using a query for each individual button, you can query ones for all buttons using the .btn class. This will save you a lot of lines. Also if you want to create even shorter code, have a look at the Object.keys() method. Implementing this will make you able to check the state for every item in a few lines.

Reply:
I just saw that you used this in your last iteration as well. Try to apply the Object.keys() here as well 💯